flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/AvroRowDeserializationSchema.java

```diff
@@ -157,13 +150,13 @@ public AvroRowDeserializationSchema(String avroSchemaString) {
 		record = new GenericData.Record(schema);
 		datumReader = new GenericDatumReader<>(schema);
 		inputStream = new MutableByteArrayInputStream();
-		decoder = DecoderFactory.get().binaryDecoder(inputStream, null);
 	}
 
 	@Override
 	public Row deserialize(byte[] message) throws IOException {
 		try {
 			inputStream.setBuffer(message);
+			Decoder decoder = DecoderFactory.get().binaryDecoder(inputStream, null);
```

Comment:
AvroRowDataDeserializationSchema should also be considered.

Reply:
Do you known the overhead of `DecoderFactory.get().binaryDecoder(inputStream, null);`?  
If it's heavy, maybe we can get a new decoder when encounters a schema change exception?

Reply:
@libenchao I've added the similar fix for `AvroRowDataDeserializationSchema `.  Also reconsidered the overhead of `DecoderFactory.get().binaryDecoder(inputStream, null)`, it would allocate 8KB buffer for each creation if don't reuse, So I changed to reuse the decode to fix this. Thanks.